core/common/src/main/java/alluxio/exception/PreconditionMessage.java

```diff
@@ -74,29 +70,14 @@
   // SEMICOLON! minimize merge conflicts by putting it on its own line
   ;
 
-  private final MessageFormat mMessage;
+  private final String mMessage;
 
   PreconditionMessage(String message) {
```

Comment:
I think I prefer option 1, since the boilerplate is restricted to one class and option 3 seems like abusing interfaces.